.maint/update_zenodo.py

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env python3
+"""Update and sort the creators list of the zenodo record."""
+import sys
+from pathlib import Path
+import json
+from fuzzywuzzy import fuzz, process
+
+# These ORCIDs should go last
+CREATORS_LAST = ['Poldrack, Russell A.', 'Gorgolewski, Krzysztof J.']
+CONTRIBUTORS_LAST = ['Ghosh, Satrajit S.']
+
+
+def sort_contributors(entries, git_lines, exclude=None, last=None):
+    """Return a list of author dictionaries, ordered by contribution."""
+    last = last or []
+    sorted_authors = sorted(entries, key=lambda i: i['name'])
+
+    first_last = [' '.join(val['name'].split(',')[::-1]).strip()
+                  for val in sorted_authors]
+    first_last_excl = [' '.join(val['name'].split(',')[::-1]).strip()
+                       for val in exclude or []]
+
+    unmatched = []
+    author_matches = []
+    position = 1
+    for ele in git_lines:
+        matches = process.extract(ele, first_last, scorer=fuzz.token_sort_ratio,
+                                  limit=2)
+        # matches is a list [('First match', % Match), ('Second match', % Match)]
+        if matches[0][1] > 80:
+            val = sorted_authors[first_last.index(matches[0][0])]
+        else:
+            # skip unmatched names
+            if ele not in first_last_excl:
+                unmatched.append(ele)
+            continue
+
+        if val not in author_matches:
+            val['position'] = position
+            author_matches.append(val)
+            position += 1
+
+    names = {' '.join(val['name'].split(',')[::-1]).strip() for val in author_matches}
+    for missing_name in first_last:
+        if missing_name not in names:
+            missing = sorted_authors[first_last.index(missing_name)]
+            missing['position'] = position
+            author_matches.append(missing)
+            position += 1
+
+    all_names = [val['name'] for val in author_matches]
+    for last_author in last:
+        author_matches[all_names.index(last_author)]['position'] = position
+        position += 1
+
+    author_matches = sorted(author_matches, key=lambda k: k['position'])
+
+    return author_matches, unmatched
+
+
+def get_git_lines(fname='line-contributors.txt'):
```

Comment:
Is this preferable to commit count? It's extremely sensitive to refactors.

Reply:
Hard to tell, I'd try this to start with and then reconsider down the line.

Reply:
FWIW, we decided to go by commit count in PyBIDS in https://github.com/bids-standard/pybids/issues/299. There wasn't a very strong objection to lines, but https://github.com/bids-standard/pybids/issues/299#issuecomment-442266549 demonstrates the issue.

Reply:
Unless we convert to "squash and merge" for PRs, commits will also just be sensitive to how modular people are in their commits... Also no strong preference on which way to go, for now ! Just pinging in the other perspective.

Reply:
Sure, that is the obvious counter, but excessive commits from one person can improve their own score, but can't diminish other people's.